### PR TITLE
JENKINS-60921 Typography changes

### PR DESCRIPTION
**Pull request by fqueiruga**
See [JENKINS-60921](https://issues.jenkins-ci.org/browse/JENKINS-60921).

This PR aims to modernize and normalize the base typographic settings of Jenkins. The main goals are:
- Move away from old and custom fonts and use the native fonts for each browser / OS combo (i.e. Segoe UI on Windows). This has the benefit of fitting  the user OS while reducing the frontend payload.
- Normalize font sizes towards a 14px font size for normal body copy. This size is the most commonly used on tools similar to Jenkins. Some UI elements may use 12px or 16px font sizes for contrast (i.e. form help messages).
- Default line height has been upped to 1.5 times the font size, in order to improve legibility.
- Update the headings hierarchy (see below). Headings have a 1.2 line height. Add CSS classes that mimic the heading styles so that they can be reused across the application.
- Stop using bootstrap 3 on Jenkins core. Wasn't removed because some plugins may depend on it.

This PR does no:
- Change font sizes for the setup wizard. The reason is that its layout is brittle and risk breaking. Small fixes to account for line height were done.
- Make font sizes responsive.
- Update form section headers.
- Rethink form sizes, using bigger forms in some UI elements.

Some of these items fall under a future iteration on [JENKINS-61367](https://issues.jenkins-ci.org/browse/JENKINS-61367).

This PR can be tested using the following docker image:
`docker run --rm -ti -p 8080:8080 -e ID=4548 jenkins/core-pr-tester`

#### TODO:
- [x] Double check that it works on IE11

#### Headings hierarchy
<details>
<summary>Click to view</summary>

- H1: 32px
- H2: 28px
- H3: 24px
- H4: 20px
- H5: 16px
- H6: 14px
</details>

### Screenshots
<details>
<summary>Click to view</summary>

Old home page
home_old

New home page, OSX
home_new_osx

New home page, Ubuntu
home_new_ubuntu

Old job page
job_old

New job page, OSX
job_new_osx

New job page, Ubuntu
job_new_ubuntu

Old job form
form_old

New job form, OSX
form_new_osx

New job form, Ubuntu
form_new_ubuntu

Old security config
form_2_old

New security config, OSX
form_2_new_osx

New security config, Ubuntu
form_2_new_ubuntu

Old plugins table
plugins_old

New plugins table, OSX
plugins_new_osx

New plugins table, Ubuntu
plugins_new_ubuntu

</details>

### Proposed changelog entries

* JENKINS-60921 Jenkins now uses modern system fonts provided by the browser when possible.
* JENKINS-60921 Changes in font size for body copy and headings to improve consistency and legibility.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] JIRA issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@halkeye 
@timja 
@oleg-nenashev 
@LinuxSuRen 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

**Review by timja — APPROVED** [review]
Couple of questions on TODOs,
Rest looks good to me,

I've tested:

Freestyle job,
Most pages in manage
configure
blue ocean
a few plugins, slack, jcasc

File: war/src/main/less/base/style.less [hunk 1]
```diff
@@ -617,7 +614,10 @@ div.behavior-loading {
     left:0; right:0;
     width:100%;
     height:100%;
-    background-color: #e4e4e4; text-align: center; font-size: 300%;
+    background-color: #e4e4e4;
+    text-align: center;
+    // TODO: use a display font, maybe 3rem
```
[the remark this answers is not in the document]
Comment (fqueiruga) [reply war/src/main/less/base/style.less]: I probably should've done it on this PR, but I don't know how to run this code  (the `div .behavior-loading` element).

File: war/src/main/js/widgets/add/addform.less [hunk 2]
```diff
@@ -77,9 +79,7 @@
 
     label {
       position: relative;
-      display: block;
-      font-size: 22px;
-      font-weight: bold;
+      // TODO: use default margin
```
[the remark this answers is not in the document]
Comment (fqueiruga) [reply war/src/main/js/widgets/add/addform.less]: Probably not, removing it!

File: war/src/main/less/base/style.less (hunk 1 above)
[the remark this answers is not in the document]
Comment (halkeye) [reply war/src/main/less/base/style.less]: I would bet the people page has it. Or top level build history. I can confirm later though

**Review by LinuxSuRen — APPROVED** [review]
LGTM

**Review by res0nance — APPROVED** [review]
LGTM, this looks like its not gated like the new UI but the screenshots seem to be of the new refreshed UI. @timja did you test this with the existing UI?

**Comment by timja** [comment]
I used both, mostly the existing ui, yes it’s not gated, not needed imo

File: war/src/main/less/pluginSetupWizard.less [hunk 3]
```diff
@@ -947,3 +943,15 @@ body .plugin-setup-wizard { // need specificity, revisit the CSS inclusion order
...
+    // simple reset
+    margin: initial;
+    padding: initial;
+}
+
+.bootstrap-3 .categories,
+.bootstrap-3 .plugins {
+    padding-left: 0;
+    padding-right: 0;
+}
```
Comment (alecharp) [inline war/src/main/less/pluginSetupWizard.less]:
```suggestion
}

```

**Review by alecharp — APPROVED** [review]
I was afraid about the header class naming, but as the tags and the classes define the same behaviour, it seems to be all good.

File: core/src/main/resources/jenkins/model/Jenkins/manage.jelly [hunk 4]
```diff
@@ -29,7 +29,6 @@ THE SOFTWARE.
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
 <l:layout title="${%Manage Jenkins}" permissions="${app.MANAGE_AND_SYSTEM_READ}">
   <l:header>
-     <link rel="stylesheet" href="${resURL}/bootstrap/css/bootstrap.min.css" type="text/css" />
```
Comment (varyvol) [inline core/src/main/resources/jenkins/model/Jenkins/manage.jelly]: Is this related to fonts changes?

File: war/src/main/less/base/typography.less [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+body,
+p {
+  font-family: @font-family-sans;
+  font-size: @font-size-sm;
+  line-height: @line-height-base;
+  color: @text-color;
+}
+
+@media screen and (min-width: 1600px) {
```
Comment (varyvol) [inline war/src/main/less/base/typography.less]: Why are these repeated and empty?

File: core/src/main/resources/jenkins/model/Jenkins/manage.jelly (hunk 4 above)
Comment (fqueiruga) [reply core/src/main/resources/jenkins/model/Jenkins/manage.jelly]: Yep, bootstrap 3 was not actually being used and it has risks of messing with default font settings.

File: war/src/main/less/base/typography.less (hunk 5 above)
Comment (fqueiruga) [reply war/src/main/less/base/typography.less]: I guess they were a placeholder for responsive typography, that I didn't do in the end. I'll remove them.

**Review by varyvol — APPROVED** [review]

**Review by MRamonLeon — APPROVED** [review]
I've played a bit with it and looks great, thanks

**Review by oleg-nenashev — APPROVED** [review]
I do not see any issues in the suggested code change, and the screenshots in the new version look better. So +1 from me, thanks @fqueiruga !

**Comment by timja** [comment]
Thanks @fqueiruga, let's merge it after 24 hours, if there's no negative feedback.

**Review by batmat — APPROVED** [review]
\o/

**Comment by batmat** [comment]
24 hours passed :fireworks: :champagne:

**Comment by timja** [comment]
> 24 hours passed 🎆 🍾
> 
> 

on the dot lol
